### ISSUE TITLE
--timeout flag doesn't behave as expected

### ISSUE THREAD

**Issue by shaunmulligan** [resin-os/resin-device-toolbox#68]
* `rdt` version: 0.8
* Operating system and architecture: OS X 10.11.5

**Current behaviour:** The command `rdt scan --timeout=120` runs and only reports the results after 2 minutes.

**Expected Behaviour:** I would expect it to report any devices as soon as it finds then, but continue scanning as well for additional devices.

**Comment by hedss** [comment]
@shaunmulligan OK, that makes sense; we can implement this by just dividing the time into a set of cumulative timeouts set to the passed argument. But, this will need some granularity. Display new results every 5 seconds (which I think was the original timeout)?

**Comment by shaunmulligan** [comment]
@hedss yeah that sounds sane. Its goes without saying this is not high priority, but nice to have :)
